### PR TITLE
Negative number filters

### LINKED ISSUE #3276: Negative Label Filters
We don't support label filtering with negative number, or duration nor bytes.

For instance this fails:

`| json | temperature > -30`

I suspect some lexer issue is going on.

### PR DESCRIPTION
**Pull request by dannykopping**
**What this PR does / why we need it**:
Implements filtering on negative numbers

**Which issue(s) this PR fixes**:
Fixes #3276

**Special notes for your reviewer**:
There was a slight change in the error handling behaviour of the parser relating to my change (see IP filtering tests), I'm not sure why just yet.

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

**Comment by cyriltovena** [comment]
Can we use ragel to fix ? https://github.com/cyriltovena/loki/tree/token/pkg/logql/log/pattern

File: pkg/logql/parser_test.go [hunk 1]
```diff
@@ -2522,6 +2522,80 @@ func TestParse(t *testing.T) {
...
+				left: newMatcherExpr([]*labels.Matcher{{Type: labels.MatchEqual, Name: "app", Value: "foo"}}),
+				pipeline: MultiStageExpr{
+					&LabelFilterExpr{
+						LabelFilterer: log.NewNumericLabelFilter(log.LabelFilterEqual, "temp", 0),
+					},
+				},
+			},
+		},
+
+		//// TODO try get this case to work
```
Comment (kavirajk) [inline pkg/logql/parser_test.go:2594]: Are you planning to support this case in this PR?

**Review by kavirajk — APPROVED** [review]
LGTM! 👍

**Comment by kavirajk** [comment]
Closing this as no activities for long time. Feel free to send new PR if anyone want's to revive the work

**Comment by slim-bean** [comment]
oh man, we keep forgetting this one, but we should fix this. I'm gonna reopen this because I really think we should merge this.

**Comment by trevorwhitney** [comment]
@slim-bean @dannykopping @kavirajk  what's the status of this PR? is this something we're ready to merge, or might it best be converted to an issue for tracking to fix at a later date?

**Comment by MasslessParticle** [comment]
We haven't touched this in ages. Is there a better way to get visibility on it? 

I'm going to close it for now.

**Pull request closed without merging**